nhrpd/nhrp_route.c

```diff
@@ -412,6 +413,35 @@ void nhrp_send_zebra_configure_arp(struct interface *ifp, int family)
 	zclient_send_message(zclient);
 }
 
+void nhrp_send_zebra_gre_source_set(struct interface *ifp,
+				    unsigned int link_idx,
+				    vrf_id_t link_vrf_id)
+{
+	struct stream *s;
+
+	if (!zclient || zclient->sock < 0) {
+		zlog_err("%s : zclient not ready", __func__);
+		return;
+	}
+	if (link_idx == IFINDEX_INTERNAL ||
+	    link_vrf_id == VRF_UNKNOWN) {
+		zlog_err("%s : link interface not set. cancel zclient cmd",
```

Comment:
error  - should be a debug?

Reply:
there are cases where you dont know the link vrf identifier, like when you run zebra in vrflite mode and you don't see where gre interfaces are sit (cross-netns interfaces).
today it is forbidden to go further.
tomorrow it will be authorised, provided that you give appropriate 'tunnel source IP address' which is not what is given today in cli.

Reply:
sorry - my question as usual is: what could an operator do with this error message? I think if you have a concrete answer, you should assign a log message id, include help/recovery text, and make the text of the error message actually say something user-oriented. If not, if there's nothing that an operator can do, and if you want to include for example a function name, this really shouldn't be a raw error.